### PR TITLE
implement DF ViewLoadout API

### PR DESCRIPTION
**Pull request by Numynum**

### CODE REVIEW

File: .devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -10,3 +10,4 @@ services:
       - ..:/workspaces/wowless
     tty: true
     working_dir: /workspaces/wowless
+    user: vscode
```
Comment (ferronn-dev) [inline .devcontainer/docker-compose.yml:13]: Out of curiosity, what does this achieve?

File: .luacheckrc [hunk 2]
```diff
@@ -37,6 +37,12 @@ files['addon/Wowless'] = {
     'SendSystemMessage',
   },
 }
+files['wowless/env.lua'] = {
+  ignore = {
+    '211/_G',
```
Comment (ferronn-dev) [inline .luacheckrc:42]: Inline luacheck comments would be better for this I think.

File: data/state/Talents.yaml [hunk 3]
```diff
@@ -4,3 +4,11 @@ value:
...
   activeConfigID_comment: strictly speaking should be spec specific (nil if no spec
     selected)
+  pendingViewLoadoutLevel: null
+  pendingViewLoadoutLevel_comment: currently unused
+  pendingViewLoadoutSpecID: null
+  viewLoadoutDataImported: false
+  viewLoadoutDataImported_comment: true after ViewLoadout is called, reset on InitializeViewLoadout
+    currently used to generate warnings for bad usage. This deviates from how the
+    real client handles it!
+  viewLoadoutSpecID: null
```
Comment (ferronn-dev) [inline data/state/Talents.yaml:14]: So... yaml "null" is weird. I'm not sure if it ends up as Lua nil. Maybe just have a comment field with all these fields documented. This situation needs to be better :-/

File: wowless/env.lua [hunk 4]
```diff
@@ -20,10 +20,40 @@ local function init(api, loader, lite)
...
+  end
+
   api.env.__wowless = {
+    debug = wowlessDebug,
     dump = dump(api),
     lite = lite,
     product = api.product,
+    quit = function(exitCode)
+      api.log(1, 'Bye!')
+      os.exit(exitCode)
```
Comment (ferronn-dev) [inline wowless/env.lua:55]: maybe `exitCode or 1`

**Review by ferronn-dev — CHANGES_REQUESTED** [review]
Thanks for doing this!

File: .devcontainer/docker-compose.yml (hunk 1 above)
Comment (Numynum) [reply .devcontainer/docker-compose.yml:13]: sets the default user when executing commands via `docker-compose exec` (as opposed to root)

File: data/state/Talents.yaml (hunk 3 above)
Comment (Numynum) [reply data/state/Talents.yaml:14]: I had `viewLoadoutSpecID:\n` before, but the auto reformatter changed it to this
I'm not a fan of leaving it out completely

File: wowless/env.lua (hunk 4 above)
Comment (Numynum) [reply wowless/env.lua:55]: Defaults to 0 atm, not sure which is more sensible 🙂

File: data/state/Talents.yaml (hunk 3 above)
Comment (ferronn-dev) [reply data/state/Talents.yaml:14]: Okay we can leave it this way for now and iterate later.

File: wowless/env.lua (hunk 4 above)
Comment (ferronn-dev) [reply wowless/env.lua:55]: I think 1 is preferable; wouldn't want to accidentally leave a quit in framexml and start passing local tests.

Comment (Numynum) [reply wowless/env.lua:55]: fair enough 😉

**Review by ferronn-dev — APPROVED** [review]

**Pull request merged**